File: benchmark/corpus/python/cmdi_vuln_1.py
"""VULN: CWE-78 — OS command injection via os.system() with user input."""
import os
import flask

app = flask.Flask(__name__)


@app.route("/ping")
def ping():
    host = flask.request.args.get("host", "localhost")
    result = os.system("ping -c 1 " + host)
    return str(result)

File: benchmark/corpus/javascript/cmdi_vuln_1.js
// VULN: CWE-78 — Command injection via exec() with unsanitised user input
const { exec } = require('child_process');
const express = require('express');
const app = express();

app.get('/ping', (req, res) => {
    const host = req.query.host;
    exec('ping -c 1 ' + host, (error, stdout, stderr) => {
        res.send(stdout || stderr);
    });
});

app.listen(3000);
